frappe/core/doctype/user/user.py

```diff
@@ -75,6 +75,7 @@ def validate(self):
 			self.validate_email_type(self.email)
 			self.validate_email_type(self.name)
 		self.add_system_manager_role()
+		self.validate_roles()
```

Comment:
let's call this something else. `self.populate_roles_from_role_profile()` maybe.

Reply:
Updated function name from `validate_roles` to  `populate_roles_from_role_profile`